### PR TITLE
chore: read `ESLINT_MOCHA_TIMEOUT` env var in Makefile.js

### LINKED ISSUE #15582: Change Request: Increase unit test timeout in Makefile.js
### ESLint version

the `main` branch

### What problem do you want to solve?

Currently some unit tests timeout on slow machines, with the following output.

```
Error: Timeout of 10000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
```

Specifically, ~30 tests fail on both SiFive Unmatched RISC-V development board and `qemu-riscv64`.

### What do you think is the correct solution?

Increase the timeout from 10s to 60s will solve.

https://github.com/eslint/eslint/blob/355b23d0c4e050be4e53292f552a47c10ec6e00e/Makefile.js#L81-L82

### Participation

- [X] I am willing to submit a pull request for this change.

### Additional comments

_No response_

### PR DESCRIPTION
**Pull request by piggynl**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[x] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

This PR will close #15582.

#### What changes did you make? (Give an overview)

It allows `npm test` to override the default mocha tests timeout of 10000ms by reading `ESLINT_MOCHA_TIMEOUT` environment variable.

### CODE REVIEW

**Comment by mdjermanovic** [comment]
I'll close-reopen to see if that will fix the CLA check.

**Comment by mdjermanovic** [comment]
Okay, the check works now. @piggynl can you sign the CLA now, there is a "Please click here to be authorized" link in [previous post](https://github.com/eslint/eslint/pull/15626#issuecomment-1046905288).

**Comment by piggynl** [comment]
Hi @mdjermanovic, I've signed the CLA now.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

I'll leave this open to see if other team members agree with this change.

**Comment by mdjermanovic** [comment]
Note to the merger: the commit tag should be `chore`.

**Review by snitin315 — APPROVED** [review]
LGTM. Could be useful in a CI environment too.

File: docs/developer-guide/unit-tests.md [hunk 1]
```diff
@@ -37,3 +37,7 @@ Running individual tests is useful when you're working on a specific bug and ite
 ## More Control on Unit Testing
 
 `npm run test:cli` is an alias of the Mocha cli in `./node_modules/.bin/mocha`. [Options](https://mochajs.org/#command-line-usage) are available to be provided to help to better control the test to run.
+
+The default timeout for tests in `npm test` is 10000ms. You may change the timeout by providing `ESLINT_MOCHA_TIMEOUT` environment variable, for example:
+
+    ESLINT_MOCHA_TIMEOUT=20000 npm test
```
Comment (snitin315) [inline docs/developer-guide/unit-tests.md:43]: Like the rest of the commands?

```suggestion
    `ESLINT_MOCHA_TIMEOUT=20000 npm test`
```

Comment (snitin315) [reply docs/developer-guide/unit-tests.md:43]: Thanks!

**Comment by piggynl** [comment]
Thanks!

**Pull request merged**